Source/WPEFramework/PluginServer.h

```diff
@@ -2814,8 +2814,6 @@ namespace PluginHost {
             {
                 _notificationLock.Lock();
 
-                ASSERT(std::find(_channelObservers.begin(), _channelObservers.end(), sink) == _channelObservers.end());
```

Comment:
Hopefully this is a temporary thing? Why do we want to allow a sink to be registered more than twice ?

Reply:
Yes, hopefully.

There is a slight problem, e.g. reload UI page.: there is a new controller.1.register call that will call Subscribe which will Register the channel observer. Obviously in this scenario there is no controller.1.unregister called. The previous connection is closed shortly after, but late enough to trigger this assert. The observer will be released on behalf of the closed channel anyway thus there's no leak. However this is now prone to a dumb programmer. 

The simplest solution would be registering the observer upfront, not doing it lazily until there  is a first Subscribe call, but that smells of a resource waste.

Reply:
The assert was reinstated.  The comment above does not apply.